acvm-repo/acir/src/circuit/mod.rs

```diff
@@ -125,21 +126,54 @@ impl Circuit {
         PublicInputs(public_inputs)
     }
 
-    pub fn write<W: std::io::Write>(&self, writer: W) -> std::io::Result<()> {
```

Comment:
The issue of having all of these serialization methods for different contexts can be solved in another issue. At least with this PR, all of the logic is in one place

Reply:
Opened this issue: https://github.com/noir-lang/noir/issues/3347